src/lib/utils/options/readers/tsconfig.ts

```diff
@@ -82,7 +82,7 @@ export class TSConfigReader extends OptionsComponent
             delete compilerOptions[key];
         }
 
-        _.merge(event.data, compilerOptions);
+        _.defaultsDeep(event.data, compilerOptions);
```

Comment:
I'm not sure why it'd be future proof? Wouldn't it mean odd usages of compiler options will break because of merging instead of overriding now? E.g. `paths` would deep merge now when you might be overriding?